subsys/net/lib/lwm2m/lwm2m_rw_plain_text.c

```diff
@@ -219,20 +224,20 @@ static size_t get_s32(struct lwm2m_input_context *in, int32_t *value)
 	return len;
 }
 
-static size_t get_s64(struct lwm2m_input_context *in, int64_t *value)
+static int get_s64(struct lwm2m_input_context *in, int64_t *value)
 {
 	return plain_text_read_int(in, value, true);
 }
 
-static size_t get_string(struct lwm2m_input_context *in,
-			 uint8_t *value, size_t buflen)
+static int get_string(struct lwm2m_input_context *in, uint8_t *value,
+		      size_t buflen)
 {
 	uint16_t in_len;
 
 	coap_packet_get_payload(in->in_cpkt, &in_len);
 
 	if (in_len > buflen) {
-		/* TODO: generate warning? */
+		LOG_WRN("Buffer too small to accommodate string, truncating");
```

Comment:
Should this be `LOG_ERR()` as we actually failed to generate the correct content.

Reply:
I can increse log severity level, it might make sense.